p2p/switch.go

```diff
@@ -92,10 +93,14 @@ type Switch struct {
 	metrics *Metrics
 }
 
-// NetAddress returns the address the switch is listening on.
+// NetAddress returns the first address the switch is listening on,
+// or nil if no addresses are found.
 func (sw *Switch) NetAddress() *NetAddress {
-	addr := sw.transport.NetAddress()
-	return &addr
+	endpoints := sw.transport.Endpoints()
+	if len(endpoints) == 0 {
+		return nil
+	}
+	return endpoints[0].NetAddress()
```

Comment:
Any particular reason for the 1st address?

Reply:
The current P2P stack only supports a single address, i.e. the first. Supporting multiple addresses will require protocol changes.